test/e2e/multihoming.go

```diff
@@ -636,6 +640,103 @@ var _ = Describe("Multi Homing", func() {
 			),
 		)
 
+		Context("localnet OVN-K secondary network", func() {
+			const (
+				clientPodName          = "client-pod"
+				nodeHostnameKey        = "kubernetes.io/hostname"
+				servicePort            = 9000
+				dockerNetworkName      = "underlay"
+				underlayServiceIP      = "60.128.0.1"
+				secondaryInterfaceName = "eth1"
+			)
+
+			var netConfig networkAttachmentConfig
+			var nodes []v1.Pod
+			var underlayBridgeName string
+			var cmdWebServer *exec.Cmd
+
+			Context("with a service running on the underlay", func() {
+				BeforeEach(func() {
+					netConfig = newNetworkAttachmentConfig(
+						networkAttachmentConfigParams{
+							name:         secondaryNetworkName,
+							namespace:    f.Namespace.Name,
+							vlanID:       localnetVLANID,
+							topology:     "localnet",
+							cidr:         secondaryLocalnetNetworkCIDR,
+							excludeCIDRs: []string{underlayServiceIP + "/32"},
+						})
+
+					By("setting up the localnet underlay")
+					nodes = ovsPods(cs)
+					Expect(nodes).NotTo(BeEmpty())
+					Expect(setupUnderlay(nodes, secondaryInterfaceName, netConfig)).To(Succeed())
+				})
+
+				BeforeEach(func() {
+					By("adding IP to the underlay docker bridge")
+					cli, err := client.NewClientWithOpts(client.FromEnv)
+					Expect(err).NotTo(HaveOccurred())
+
+					gatewayIP, err := getNetworkGateway(cli, dockerNetworkName)
+					Expect(err).NotTo(HaveOccurred())
+
+					underlayBridgeName, err = findInterfaceByIP(gatewayIP)
+					Expect(err).NotTo(HaveOccurred())
+
+					cmd := exec.Command("sudo", "ip", "addr", "add", underlayServiceIP+"/24", "dev", underlayBridgeName)
+					cmd.Stderr = os.Stderr
+					err = cmd.Run()
+					Expect(err).NotTo(HaveOccurred())
+				})
+
+				BeforeEach(func() {
+					By("starting a service, connected to the underlay")
+					cmdWebServer = exec.Command("python3", "-m", "http.server", "--bind", underlayServiceIP, strconv.Itoa(servicePort))
+					cmdWebServer.Stderr = os.Stderr
+					err := cmdWebServer.Start()
+					Expect(err).NotTo(HaveOccurred(), "failed to create web server, port might be busy")
```

Comment:
nit: you can inline these, which allows for positive logic - i.e. `Expect(cmdWebServer.Start()).To(Succeed())`